File: BungieNetApi/Models/DestinyActivityDifficultyTier.cs
using System.Runtime.Serialization;

namespace GhostSharper.Models
{
    /// <summary>
    /// An enumeration representing the potential difficulty levels of an activity. Their names are... more qualitative than quantitative.
    /// </summary>
    
    public enum DestinyActivityDifficultyTier
    {
        /// <summary>
        /// Trivial = 0
        /// </summary>
        [EnumMember(Value = "0")]
        Trivial = 0,

        /// <summary>
        /// Easy = 1
        /// </summary>
        [EnumMember(Value = "1")]
        Easy = 1,

        /// <summary>
        /// Normal = 2
        /// </summary>
        [EnumMember(Value = "2")]
        Normal = 2,

        /// <summary>
        /// Challenging = 3
        /// </summary>
        [EnumMember(Value = "3")]
        Challenging = 3,

        /// <summary>
        /// Hard = 4
        /// </summary>
        [EnumMember(Value = "4")]
        Hard = 4,

        /// <summary>
        /// Brave = 5
        /// </summary>
        [EnumMember(Value = "5")]
        Brave = 5,

        /// <summary>
        /// AlmostImpossible = 6
        /// </summary>
        [EnumMember(Value = "6")]
        AlmostImpossible = 6,

        /// <summary>
        /// Impossible = 7
        /// </summary>
        [EnumMember(Value = "7")]
        Impossible = 7,

    }
}

File: pkg-generated/models/DestinyActivityDifficultyTier.go
package bungieAPI

// An enumeration representing the potential difficulty levels of an activity. Their names
// are... more qualitative than quantitative.
type DestinyActivityDifficultyTier int

const (
	DestinyActivityDifficultyTierTrivial          = 0
	DestinyActivityDifficultyTierEasy             = 1
	DestinyActivityDifficultyTierNormal           = 2
	DestinyActivityDifficultyTierChallenging      = 3
	DestinyActivityDifficultyTierHard             = 4
	DestinyActivityDifficultyTierBrave            = 5
	DestinyActivityDifficultyTierAlmostImpossible = 6
	DestinyActivityDifficultyTierImpossible       = 7
)
